addons/mail/static/src/messaging/entity/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -298,44 +309,33 @@ function MessagingNotificationHandlerFactory({ Entity }) {
          * @param {string} data.uuid
          */
         _handleNotificationPartnerChannel(data) {
-            const {
-                channel_type,
-                id,
-                info,
-                is_minimized,
-                name,
-                state: channelState,
-            } = data;
-            if (channel_type !== 'channel' || channelState !== 'open') {
-                return;
-            }
+            const convertedData = this.env.entities.Thread.convertData(data);
+
             let channel = this.env.entities.Thread.find(thread =>
-                thread.id === id &&
+                thread.id === convertedData.id &&
                 thread.model === 'mail.channel'
             );
+            const wasCurrentPartnerMember = channel && channel.members.includes(this.env.messaging.currentPartner);
+
+            channel = this.env.entities.Thread.insert(Object.assign(
+                { isPinned: true },
+                convertedData,
+            ));
+
             if (
-                !is_minimized &&
-                info !== 'creation' &&
-                (
-                    !channel ||
-                    !channel.members.includes(this.env.messaging.currentPartner)
-                )
+                channel.channel_type === 'channel' &&
+                channel.foldState === 'open' &&
+                data.info !== 'creation' &&
+                !wasCurrentPartnerMember
             ) {
-                this.env.do_notify(
+                this.do_notify(
                     this.env._t("Invitation"),
-                    _.str.sprintf(
-                        this.env._t("You have been invited to: %s"),
-                        name
-                    )
+                    this.env._t("You have been invited to: ") + channel.name
```

Comment:
also %s should be kept

I just badly copied the old js over the new one that was better